### PR TITLE
excercises 2

### PR DESCRIPTION
**Pull request by LuisRG98**
new exercise

### CODE REVIEW

File: implement_strstr().py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Solution:
+    def strStr(self, haystack: str, needle: str) -> int:
+        if needle in haystack:
+            return(haystack.index(needle))
+        else:
+            return -1
```
Comment (reddytocode) [inline implement_strstr().py]: This is a common thing that can be refactored
```suggestion
        if needle in haystack:
            return haystack.index(needle)
        return -1
```

Comment (reddytocode) [reply implement_strstr().py]: they can give you something like what you did, and ask you: What can be improved? or What is wrong?

File: number_of_1_bits.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Solution:
+    def hammingWeight(self, n: int) -> int:
```
Comment (reddytocode) [inline number_of_1_bits.py:2]: I know that the name of this method was a given one from whatever platform you're using to train your algorithms, but at some point, you need to follow the best practices dictated in this case by PEP8 https://peps.python.org/pep-0008/

Comment (reddytocode) [reply number_of_1_bits.py:2]: method or functions should use underscores to split words

File: single_number.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Solution:
+    def singleNumber(self, nums: List[int]) -> int:
+        for i in nums:
+            if nums.count(i)==1:
```
Comment (reddytocode) [inline single_number.py:4]: this is equivalent to if mums.count(i) 
0 => False
any value => True

File: valid_palindrome.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Solution:
+    def isPalindrome(self, s: str) -> bool:
+        for i in s:
+            if not i.isalnum():
+                s = s.replace(i,"")
+        
+        s2 = s.lower()
```
Comment (reddytocode) [inline valid_palindrome.py:7]: nice use of string methods, but just as a recommendation, look at how you can achieve the same without builtin functions (I do that just for curiosity)

**Review by reddytocode — APPROVED** [review]

File: implement_strstr().py (hunk 1 above)
Comment (LuisRG98) [reply implement_strstr().py]: thanks for the clarification

**Pull request merged**